### ISSUE TITLE
[enhancement] Full width split breaks offsets

### ISSUE THREAD

**Issue by nullchilly** [akinsho/bufferline.nvim#429]
Don't update bufferline offset if new buffer is terminal?

https://user-images.githubusercontent.com/56817415/169694318-3cfc84db-e8ff-4184-becf-5664e7f35028.mp4

**Comment by akinsho** [comment]
@nullchilly I'm not exactly sure what I'm seeing or what exactly the context is for this report i.e. config, nvim version reproduction setups etc. Either way, I recommend that you set your terminal buffers to be `nobuflisted`

**Comment by akinsho** [comment]
Also, a more clear issue title would be much appreciated, I can see something flickering, but that isn't really a good amount of information. The chances of me fixing any issue with this little information is close to zero unfortunately

**Comment by nullchilly** [comment]
The tiny flickering is from [stabilizer.nvim](https://github.com/luukvbaal/stabilize.nvim), I will figure it out later.

[My config](https://github.com/nullchilly/dots/blob/main/.config/nvim/lua/plugins.lua#L146) uses bufferline and toggleterm with  neovim 0.7

Not sure how to put terminal buffers to `nobuflisted`? I skeemed through your config and didn't find anything related to it

**Comment by akinsho** [comment]
@nullchilly please read `:help buflisted`, regarding the config, I don't really have time to read through each user who reports an issue's dotfiles. If you can narrow it down to one minimal init.lua file that triggers the same problem, I can test. This can maybe just be resolved by setting no buflisted though e.g. `autocmd! term://* setlocal nobuflisted`

**Comment by nullchilly** [comment]
```lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvt-min/site]]
local package_root = "/tmp/nvt-min/site/pack"
local install_path = package_root .. "/packer/start/packer.nvim"
local function load_plugins()
  require("packer").startup {
    {
      "wbthomason/packer.nvim",
      "kyazdani42/nvim-tree.lua",
      "kyazdani42/nvim-web-devicons",
			"akinsho/bufferline.nvim",
			"akinsho/toggleterm.nvim"
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. "/plugin/packer_compiled.lua",
      display = { non_interactive = true },
    },
  }
end
if vim.fn.isdirectory(install_path) == 0 then
  print "Installing nvim-tree and dependencies."
  vim.fn.system { "git", "clone", "--depth=1", "https://github.com/wbthomason/packer.nvim", install_path }
end
load_plugins()
require("packer").sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua setup()]]
vim.opt.termguicolors = true
vim.opt.cursorline = true

_G.setup = function()
  require'nvim-tree'.setup {}
	require'bufferline'.setup{ options = { offsets = { { filetype = "NvimTree", } }, } }
	require'toggleterm'.setup{}
end
```

**Comment by nullchilly** [comment]
I suppose it is like this?

```lua
vim.api.nvim_create_autocmd("TermOpen", {
	pattern = "term://*",
	callback = function() vim.opt_local.buflisted = false end,
})
```

I added it to `nvim -u min.lua` and `:execute'NvimTreeOpen' | execute'ToggleTerm'`

The offset doesn't work

**Comment by akinsho** [comment]
Ah @nullchilly I understand the issue now. This is a known limitation of the offsets feature. Basically the offset works by reading the layout in neovim if you have a **Full length** vertical buffer and specify the filetype this will work. If you however split the Full length buffer, which is what your terminal is doing, you will break the layout detection. There are many issues about this, e.g. #113. This issue is a pain and offsets are basically a feature I added for another user that is kind of feature creep anyway in my opinion. If you want a fix, I would happily take a PR, but I don't want to go down this rabbit hole again. When I got asked to add this feature in the first place, I would have rejected the request if I had known how many problems it would create for me 🤕. Basically I suggest not making your terminal full width e.g. don use `wincmd J` just split the current window.
